runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -659,12 +613,197 @@ public synchronized CorfuRuntime connect() {
 
         checkVersion();
 
-        if (loadSmrMapsAtConnect) {
+        if (parameters.isUseFastLoader()) {
             FastObjectLoader fastLoader = new FastObjectLoader(this)
-                    .setBatchReadSize(getBulkReadSize())
-                    .setTimeoutInMinutesForLoading(timeoutInMinutesForFastLoading);
+                    .setBatchReadSize(parameters.getBulkReadSize())
+                    .setTimeoutInMinutesForLoading((int) parameters.fastLoaderTimeout.toMinutes());
             fastLoader.loadMaps();
         }
         return this;
     }
+
+    // Below are deprecated methods which should no longer used and may be deprecated in the future.
+
+    // region Deprecated Constructors
+    /**
+     * Constructor for CorfuRuntime.
+     *
+     **/
+    @Deprecated
+    public CorfuRuntime() {
+        this(CorfuRuntimeParameters.builder().build());
+    }
+
+    /**
+     * Parse a configuration string and get a CorfuRuntime.
+     *
+     * @param configurationString The configuration string to parse.
+     */
+    @Deprecated
+    public CorfuRuntime(String configurationString) {
+        this(CorfuRuntimeParameters.builder().build());
+        this.parseConfigurationString(configurationString);
+    }
+    // endregion
+
+    // region Deprecated Setters
+
+    /**
+     * Enable TLS.
+     * @deprecated  Deprecated, set using {@link CorfuRuntimeParameters} instead.
+     **/
+    @Deprecated
+    public CorfuRuntime enableTls(String keyStore, String ksPasswordFile, String trustStore,
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)